service_manager/yapapi_connector.py

```diff
@@ -0,0 +1,73 @@
+from yapapi import Golem
+import asyncio
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from typing import List, Optional, Callable, Awaitable
+    from .service_wrapper import ServiceWrapper
+    from .service_manager import ServiceManager
+
+
+class YapapiConnector():
+    def __init__(self, executor_cfg: dict, exception_handler: 'Callable[[ServiceManager, Exception], Awaitable[None]]'):
```

Comment:
`exception_handler` is called on `Exception` only.
```suggestion
    def __init__(self, executor_cfg: dict, exception_handler: 'Callable[[Exception], Awaitable[Any]]'):
```